LanguageFeatures/Records/record_type_annotations_A03_t02.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if a record type has any of:
+///
+/// The same field name more than once.
+///
+/// No named fields and only one positional field. This isn't ambiguous, since
+/// there are no parenthesized type expressions in Dart. But there is no reason
+/// to allow single positional element record types when the corresponding
+/// record values are prohibited.
+///
+/// A field named hashCode, runtimeType, noSuchMethod, or toString.
+///
+/// A field name that starts with an underscore.
+///
+/// @description Checks that it is a compile-time error if a record type has the
+/// same field name more than once. Test positional fields
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=records
+
+typedef R1 = (int i, {String s, int i});
```

Comment:
Await clarification in https://github.com/dart-lang/language/issues/2428.

Reply:
Seems that in https://github.com/dart-lang/language/issues/2428 people agreed that the behaviour supposed in these tests is the expected.. I don't see any other opened questions in this PR

Reply:
Let's decide that all the name conflicts exist, so `(int i, int i)` and `(int i, {int i})` are both errors.